src/Processors/Formats/Impl/PrometheusTextOutputFormat.cpp

```diff
@@ -107,17 +108,25 @@ void PrometheusTextOutputFormat::fixupBucketLabels(CurrentMetric & metric)
 {
     String bucket_label = metric.type == "histogram" ? "le" : "quantile";
 
-    std::sort(metric.values.begin(), metric.values.end(),
+    ::sort(metric.values.begin(), metric.values.end(),
         [&bucket_label](const auto & lhs, const auto & rhs)
         {
+            bool lhs_labels_contain_sum = lhs.labels.contains("sum");
+            bool lhs_labels_contain_count = lhs.labels.contains("count");
+            bool lhs_labels_contain_sum_and_count = lhs_labels_contain_sum || lhs_labels_contain_count;
+
+            bool rhs_labels_contain_sum = rhs.labels.contains("sum");
+            bool rhs_labels_contain_count = rhs.labels.contains("count");
+            bool rhs_labels_contain_sum_and_count = rhs_labels_contain_sum || rhs_labels_contain_count;
+
             /// rows with labels at the beginning and then `_sum` and `_count`
-            if (lhs.labels.contains("sum") && rhs.labels.contains("count"))
+            if (lhs_labels_contain_sum && rhs_labels_contain_count)
                 return true;
-            if (lhs.labels.contains("count") && rhs.labels.contains("sum"))
+            else if (lhs_labels_contain_count && rhs_labels_contain_sum)
                 return false;
-            if (rhs.labels.contains("sum") || rhs.labels.contains("count"))
+            else if (rhs_labels_contain_sum_and_count && !lhs_labels_contain_sum_and_count)
```

Comment:
Is that not just `else if (rhs_labels_contain_sum_and_count)`?

Reply:
I am not sure about this logic, lets ask @vdimir.

Reply:
Current logic is fine. It's required to order values with labels `sum` and `count` with other labels. We want to get following order `[something1, somtething2, ..., somethingn, sum, count]`, so this checks works as expected to handle this special labels:

```cpp
if (lhs_labels_contain_sum && rhs_labels_contain_count) // lhs is 'sum' < rhs is 'count'
    return true;
else if (lhs_labels_contain_count && rhs_labels_contain_sum) // lhs is 'count > rhs is 'sum'
    return false;
else if (rhs_labels_contain_sum_or_count && !lhs_labels_contain_sum_or_count) // lhs is not 'count' and not 'sum' <  rhs is 'count' or 'sum'
    return true;
else if (lhs_labels_contain_sum_or_count && !rhs_labels_contain_sum_or_count) // lhs is 'count' or 'sum' > rhs is not 'count' and not 'sum'
    return false;

```